src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+
+    /**
+     * Pointer to first node.
+     */
+    private MyLinkedList.Node<T> first;
+
+    /**
+     * Pointer to last node.
+     */
+    private MyLinkedList.Node<T> last;
```

Comment:
Don't use class Objects for `first` and `last`. Instead, you should use the inner class `Node` directly.